pymongo/collection.py

```diff
@@ -2162,15 +2163,13 @@ def drop_index(
         cmd.update(kwargs)
         if comment is not None:
             cmd["comment"] = comment
-        with self._socket_for_writes(session) as sock_info:
```

Comment:
IIUC this still needs to use _socket_for_writes because it's a write command, not a read. Also looks like this change broke a few drop_index tests.

Reply:
Fixed